cmd/vicadmin/server.go

```
@@ -114,39 +130,158 @@ func (s *server) listenPort() int {
 	return s.l.Addr().(*net.TCPAddr).Port
 }
 
+// Enforces authentication on route `link` and runs `handler` on successful auth
+func (s *server) AuthenticatedHandle(link string, h http.Handler) {
+	s.Authenticated(link, h.ServeHTTP)
+}
+
+func (s *server) Handle(link string, h http.Handler) {
+	s.mux.Handle(link, gorillacontext.ClearHandler(h))
+}
+
+// Enforces authentication on route `link` and runs `handler` on successful auth
+func (s *server) Authenticated(link string, handler func(http.ResponseWriter, *http.Request)) {
+	defer trace.End(trace.Begin(""))
+
+	authHandler := func(w http.ResponseWriter, r *http.Request) {
+		websession, err := store.Get(r, sessionCookieKey)
+		if err != nil {
+			log.Errorf("Got \"%s\" while looking up cookie store", err.Error())
+		}
+
+		if len(r.TLS.PeerCertificates) > 0 { // the user is authenticated by certificate at connection time
+			usersess := s.uss.Add("root", &rootConfig.Config)
+			websession, _ := store.Get(r, sessionCookieKey)
```

You ignore the error here but not on line 147?